File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>
#include<math.h>

int main(void)
{
    float c = 0;
    int change = 0, counter = 0;
    
    
    do
    {
        printf("How much change is owed, sir?\n");
        c = GetFloat();
    }
    while(c < 0);
    
    change = roundf(c * 100);       // in cents instead of dollars.
    
    while(change >= 25)
    {
        change = change - 25;
        counter++;
    }
    while(change >= 10)
    {
        change = change - 10;
        counter++;
    }
    while(change >= 5)
    {
        change = change - 5;
        counter++;
    }
    while(change >=1)
    {
        change = change -1;   
        counter++;
    }
    printf("%d\n", counter);
}

File: CS50_2015/pset6/greedy.py
def main():
    x = 0
    counter = 0
    while(x <= 0):
        x = float(input("How much change is owed, sir? "))

    #convert dollar to cents
    change = x*100
    
    while(change >= 25):
        change = change - 25
        counter = counter + 1
    while(change >= 10):
        change = change - 10
        counter = counter + 1
    while(change >= 5):
        change = change - 5
        counter = counter + 1
    while(change >= 1):
        change = change - 1
        counter = counter + 1
    print(counter)
    
    
    
if __name__ == "__main__":
    main()